app/controllers/courses_controller.rb

```diff
@@ -17,6 +17,14 @@ def index
   # GET /courses/1
   # GET /courses/1.json
   def show
+    session[:course_id] = @course.id
```

Comment:
Instead of using the session here, we should append the course id to the query string when they are redirected over to  the lectures controller, then it can be retrieved in the `params` object